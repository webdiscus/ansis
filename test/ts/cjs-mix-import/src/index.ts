import ansis, { Ansis, AnsiColors, AnsiStyles, red, green, blue, yellow, magenta, hex } from 'ansis';

const log = console.log;

const write = (
  content: string,
  tagColor: AnsiColors | AnsiStyles,
  bgTagColor: AnsiColors,
  tag: string,
  error = false
) => {
  const stream = error ? process.stderr : process.stdout;
  const tagLabel = `[${tag}]`;
  const item = `${ansis[bgTagColor][tagColor](tagLabel)} ${ansis.white(content)}\n`;
  stream.write(item);
};

// color names by hex: https://chir.ag/projects/name-that-color/
const pink = ansis.hex('#FF75D1').bold; // using default import
const lime = hex('#BFFF00').bold; // using named import

// create new instance
const ansis2 = new Ansis();

const boldText = ansis.bold.open + 'text bold' + ansis.bold.close;
const styledText = magenta.underline.italic('underline italic magenta');
const str = 'string';

log(ansis2.cyan('new instance'));
log(boldText);
log(pink('pink'));
log(lime('lime'));
log(red('red'));
log(green.bold('green'));
log(blue.underline.italic('blue'));
log(yellow.italic(`yellow using ${str} variable`));
log(red.bgCyan.underline.hex('#fce')('text underline'));
log(styledText);
log(ansis.strip(styledText), '<= strip ANSI codes');

// test the AnsiColors and AnsiStyles types
write('The tip message.', 'italic', 'bgGreen', 'TIP', false);
write('The information.', 'black', 'bgBlue', 'INFO', false);