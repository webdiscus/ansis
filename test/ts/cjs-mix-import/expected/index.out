[36mnew instance[39m
[1mtext bold[22m
[38;2;255;117;209m[1mpink[22m[39m
[38;2;191;255;0m[1mlime[22m[39m
[31mred[39m
[32m[1mgreen[22m[39m
[34m[4m[3mblue[23m[24m[39m
[33m[3myellow using string variable[23m[39m
[31m[46m[4m[38;2;255;204;238mtext underline[39m[24m[49m[39m
[35m[4m[3munderline italic magenta[23m[24m[39m
underline italic magenta <= strip ANSI codes
[42m[3m[TIP][23m[49m [37mThe tip message.[39m
[44m[30m[INFO][39m[49m [37mThe information.[39m
